
<!DOCTYPE html>
<html>
<head>
<style type="text/css">
.inline {
  background-color: #f7f7f7;
  border:solid 1px #B0B0B0;
}
.error {
	font-weight: bold;
	color: #FF0000;
}
.warning {
	font-weight: bold;
}
.message {
	font-style: italic;
}
.source, .output, .warning, .error, .message {
	padding: 0 1em;
  border:solid 1px #F7F7F7;
}
.source {
  background-color: #f5f5f5;
}
.left {
  text-align: left;
}
.right {
  text-align: right;
}
.center {
  text-align: center;
}
.hl.num {
  color: #AF0F91;
}
.hl.str {
  color: #317ECC;
}
.hl.com {
  color: #AD95AF;
  font-style: italic;
}
.hl.opt {
  color: #000000;
}
.hl.std {
  color: #585858;
}
.hl.kwa {
  color: #295F94;
  font-weight: bold;
}
.hl.kwb {
  color: #B05A65;
}
.hl.kwc {
  color: #55aa55;
}
.hl.kwd {
  color: #BC5A65;
  font-weight: bold;
}
</style>
  <script src="https://yihui.name/media/js/center-images.js"></script>
  <title>A Report Generated by knitr</title>
</head>
<body>

  <p>This report is automatically generated with the R
    package <a href="https://yihui.name/knitr/"><strong>knitr</strong></a>
    (version <code class="knitr inline">1.30</code>)
    .</p>

<div class="chunk" id="auto-report"><div class="rcode"><div class="source"><pre class="knitr r"><span class="hl com">#needed_packages----</span>
<span class="hl kwd">library</span><span class="hl std">(ensembldb)</span>
<span class="hl kwd">library</span><span class="hl std">(EnsDb.Hsapiens.v86)</span>
<span class="hl kwd">library</span><span class="hl std">(tximport)</span>
<span class="hl com">#getting file paths ----</span>
<span class="hl kwd">setwd</span><span class="hl std">(</span><span class="hl str">&quot;/home/minato/Desktop/itsok/omics/trancriptomics_diy/course_diy/data&quot;</span><span class="hl std">)</span>
<span class="hl std">experiment_data</span><span class="hl kwb">&lt;-</span><span class="hl kwd">data.frame</span><span class="hl std">(</span><span class="hl kwc">sample_name</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">'CL08'</span><span class="hl std">,</span><span class="hl str">'CL10'</span><span class="hl std">,</span><span class="hl str">'CL11'</span><span class="hl std">,</span><span class="hl str">'CL12'</span><span class="hl std">,</span><span class="hl str">'CL13'</span><span class="hl std">,</span>
                                          <span class="hl str">'HS01'</span><span class="hl std">,</span><span class="hl str">'HS02'</span><span class="hl std">,</span><span class="hl str">'HS03'</span><span class="hl std">,</span><span class="hl str">'HS04'</span><span class="hl std">,</span><span class="hl str">'HS05'</span><span class="hl std">),</span>
                            <span class="hl kwc">condition</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">'Disease'</span><span class="hl std">,</span><span class="hl str">'Disease'</span><span class="hl std">,</span><span class="hl str">'Disease'</span><span class="hl std">,</span><span class="hl str">'Disease'</span><span class="hl std">,</span><span class="hl str">'Disease'</span><span class="hl std">,</span>
                                        <span class="hl str">'Healthy'</span><span class="hl std">,</span><span class="hl str">'Healthy'</span><span class="hl std">,</span><span class="hl str">'Healthy'</span><span class="hl std">,</span><span class="hl str">'Healthy'</span><span class="hl std">,</span><span class="hl str">'Healthy'</span><span class="hl std">))</span>

<span class="hl kwd">setwd</span><span class="hl std">(</span><span class="hl str">&quot;/home/minato/Desktop/itsok/omics/trancriptomics_diy/course_diy/data&quot;</span><span class="hl std">)</span>
<span class="hl com">#in once shot we get the path of all tsv files we get from kallisto</span>
<span class="hl std">file_paths</span><span class="hl kwb">&lt;-</span><span class="hl kwd">file.path</span><span class="hl std">(experiment_data</span><span class="hl opt">$</span><span class="hl std">sample_name,</span><span class="hl str">'abundance.tsv'</span><span class="hl std">)</span>
<span class="hl kwd">print</span><span class="hl std">(experiment_data)</span>
</pre></div>
<div class="output"><pre class="knitr r">##    sample_name condition
## 1         CL08   Disease
## 2         CL10   Disease
## 3         CL11   Disease
## 4         CL12   Disease
## 5         CL13   Disease
## 6         HS01   Healthy
## 7         HS02   Healthy
## 8         HS03   Healthy
## 9         HS04   Healthy
## 10        HS05   Healthy
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">#Annotation----</span>
<span class="hl std">columns_avail</span><span class="hl kwb">&lt;-</span><span class="hl kwd">listColumns</span><span class="hl std">(EnsDb.Hsapiens.v86)</span>  <span class="hl com">#columns we can retrive for our annotation from human gene database</span>
<span class="hl com">#from getting information from column_avail, we put tx_id and gene_name Below</span>
<span class="hl std">trans</span><span class="hl kwb">&lt;-</span><span class="hl std">ensembldb</span><span class="hl opt">::</span><span class="hl kwd">transcripts</span><span class="hl std">(EnsDb.Hsapiens.v86,</span>
                              <span class="hl kwc">columns</span><span class="hl std">=</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">'tx_id'</span><span class="hl std">,</span><span class="hl str">'gene_name'</span><span class="hl std">))</span>
<span class="hl std">rna_ids</span><span class="hl kwb">&lt;-</span><span class="hl kwd">as.data.frame</span><span class="hl std">(trans)</span>
<span class="hl kwd">rownames</span><span class="hl std">(rna_ids)</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl num">1</span><span class="hl opt">:</span><span class="hl kwd">length</span><span class="hl std">(rna_ids</span><span class="hl opt">$</span><span class="hl std">seqnames))</span>
<span class="hl com">#we only need gene names for our transcripts generated from kallisto </span>
<span class="hl std">rna_ids</span><span class="hl kwb">&lt;-</span><span class="hl std">dplyr</span><span class="hl opt">::</span><span class="hl kwd">select</span><span class="hl std">(rna_ids,</span><span class="hl str">'tx_id'</span><span class="hl std">,</span><span class="hl str">'gene_name'</span><span class="hl std">)</span>
<span class="hl kwd">colnames</span><span class="hl std">(rna_ids)</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">'target_id'</span><span class="hl std">,</span><span class="hl str">'gene_name'</span><span class="hl std">)</span>  <span class="hl com">#we have to give the same target name as the tsv file of transcripts from kallisto</span>
<span class="hl std">genes</span><span class="hl kwb">&lt;-</span><span class="hl kwd">tximport</span><span class="hl std">(</span><span class="hl kwc">files</span> <span class="hl std">= file_paths,</span>
                <span class="hl kwc">type</span> <span class="hl std">=</span> <span class="hl str">'kallisto'</span><span class="hl std">,</span>
                <span class="hl kwc">countsFromAbundance</span> <span class="hl std">=</span> <span class="hl str">&quot;lengthScaledTPM&quot;</span><span class="hl std">,</span>
                <span class="hl kwc">tx2gene</span> <span class="hl std">= rna_ids,</span>
                <span class="hl kwc">ignoreTxVersion</span> <span class="hl std">= T)</span>
</pre></div>
<div class="message"><pre class="knitr r">## Note: importing `abundance.h5` is typically faster than `abundance.tsv`
</pre></div>
<div class="message"><pre class="knitr r">## reading in files with read_tsv
</pre></div>
<div class="message"><pre class="knitr r">## 1 2 3 4 5 6 7 8 9 10 
## transcripts missing from tx2gene: 13522
## summarizing abundance
## summarizing counts
## summarizing length
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">#re-formatting and feature engg data for normalization and visualization----</span>
<span class="hl kwd">library</span><span class="hl std">(edgeR)</span>
<span class="hl kwd">library</span><span class="hl std">(tidyverse)</span>
<span class="hl std">gene_tpm</span><span class="hl kwb">&lt;-</span><span class="hl kwd">data.frame</span><span class="hl std">(genes</span><span class="hl opt">$</span><span class="hl std">abundance)</span>
<span class="hl std">gene_count</span><span class="hl kwb">&lt;-</span><span class="hl std">genes</span><span class="hl opt">$</span><span class="hl std">counts</span>
<span class="hl std">dge_list</span><span class="hl kwb">&lt;-</span><span class="hl kwd">DGEList</span><span class="hl std">(gene_count)</span>  <span class="hl com">#to creat dge list it need raw counts not the tpm </span>
<span class="hl com">#Counts per million </span>
<span class="hl std">dge_list_log</span><span class="hl kwb">&lt;-</span><span class="hl kwd">cpm</span><span class="hl std">(dge_list,</span><span class="hl kwc">log</span><span class="hl std">=</span><span class="hl num">TRUE</span><span class="hl std">)</span>
<span class="hl std">dge_list_log</span><span class="hl kwb">&lt;-</span><span class="hl kwd">data.frame</span><span class="hl std">(</span><span class="hl str">'gene_name'</span><span class="hl std">=</span><span class="hl kwd">rownames</span><span class="hl std">(dge_list_log),dge_list_log,</span><span class="hl kwc">row.names</span> <span class="hl std">=</span> <span class="hl kwa">NULL</span><span class="hl std">)</span>
<span class="hl kwd">colnames</span><span class="hl std">(dge_list_log)</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">'gene_name'</span><span class="hl std">,experiment_data</span><span class="hl opt">$</span><span class="hl std">sample_name)</span>
<span class="hl std">dge_list_log_pivot</span><span class="hl kwb">&lt;-</span><span class="hl kwd">pivot_longer</span><span class="hl std">(</span><span class="hl kwc">data</span><span class="hl std">=dge_list_log,</span>
                                 <span class="hl kwc">cols</span> <span class="hl std">= CL08</span><span class="hl opt">:</span><span class="hl std">HS05,</span>
                                 <span class="hl kwc">names_to</span> <span class="hl std">=</span> <span class="hl str">'sample_name'</span><span class="hl std">,</span>
                                 <span class="hl kwc">values_to</span> <span class="hl std">=</span> <span class="hl str">'expression'</span><span class="hl std">)</span>
<span class="hl std">pl1</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ggplot</span><span class="hl std">(dge_list_log_pivot,</span><span class="hl kwd">aes</span><span class="hl std">(</span><span class="hl kwc">x</span><span class="hl std">=sample_name,</span><span class="hl kwc">y</span><span class="hl std">=expression))</span><span class="hl opt">+</span>
  <span class="hl kwd">geom_violin</span><span class="hl std">(</span><span class="hl kwd">aes</span><span class="hl std">(</span><span class="hl kwc">fill</span><span class="hl std">=sample_name),</span><span class="hl kwc">trim</span> <span class="hl std">= F)</span><span class="hl opt">+</span>
  <span class="hl kwd">stat_summary</span><span class="hl std">(</span><span class="hl kwc">fun</span><span class="hl std">=</span><span class="hl str">'median'</span><span class="hl std">,</span><span class="hl kwc">size</span><span class="hl std">=</span><span class="hl num">0.1</span><span class="hl std">)</span><span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl std">(</span><span class="hl kwc">x</span><span class="hl std">=</span><span class="hl str">'sample name'</span><span class="hl std">,</span><span class="hl kwc">y</span><span class="hl std">=</span><span class="hl str">'log 2 expression'</span><span class="hl std">,</span>
       <span class="hl kwc">title</span><span class="hl std">=</span><span class="hl str">'log2 count per million'</span><span class="hl std">,</span>
       <span class="hl kwc">subtitle</span> <span class="hl std">=</span> <span class="hl str">'untrimmed - no normalization'</span><span class="hl std">,</span>
       <span class="hl kwc">caption</span> <span class="hl std">=</span> <span class="hl kwd">Sys.time</span><span class="hl std">())</span><span class="hl opt">+</span>
  <span class="hl kwd">coord_flip</span><span class="hl std">()</span>
<span class="hl kwd">print</span><span class="hl std">(pl1)</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 10 rows containing missing values (geom_segment).
</pre></div>
</div><div class="rimage center"><img src="figure/diy-transcriptomics-Rhtmlauto-report-1.png" title="plot of chunk auto-report" alt="plot of chunk auto-report" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">#trimming or filtering  the data----</span>
<span class="hl com">#let's see how many rows having all the values(counts) having 0,in main dge_list</span>
<span class="hl std">zero_count_expression</span><span class="hl kwb">&lt;-</span><span class="hl std">dge_list</span><span class="hl opt">$</span><span class="hl std">counts</span><span class="hl opt">==</span><span class="hl num">0</span>
<span class="hl std">all.sample.zero.exp</span><span class="hl kwb">=</span><span class="hl kwd">rowSums</span><span class="hl std">(zero_count_expression)</span><span class="hl opt">==</span><span class="hl num">10</span>
<span class="hl kwd">table</span><span class="hl std">(all.sample.zero.exp)</span>
</pre></div>
<div class="output"><pre class="knitr r">## all.sample.zero.exp
## FALSE  TRUE 
## 29580  6015
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">#this can be done in</span>
<span class="hl kwd">table</span><span class="hl std">(</span><span class="hl kwd">rowSums</span><span class="hl std">(dge_list</span><span class="hl opt">$</span><span class="hl std">counts</span><span class="hl opt">==</span><span class="hl num">0</span><span class="hl std">)</span><span class="hl opt">==</span><span class="hl num">10</span><span class="hl std">)</span>
</pre></div>
<div class="output"><pre class="knitr r">## 
## FALSE  TRUE 
## 29580  6015
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl com">#filter the counts per million genes having greater then 1 values in 5 sample</span>
<span class="hl std">filter_genes</span><span class="hl kwb">&lt;-</span><span class="hl kwd">rowSums</span><span class="hl std">(</span><span class="hl kwd">cpm</span><span class="hl std">(dge_list)</span><span class="hl opt">&gt;</span><span class="hl num">1</span><span class="hl std">)</span><span class="hl opt">&gt;=</span><span class="hl num">5</span>
<span class="hl std">dge_list_filter</span><span class="hl kwb">&lt;-</span><span class="hl std">dge_list[filter_genes,]</span>
<span class="hl std">dge_list_filter_log</span><span class="hl kwb">&lt;-</span><span class="hl kwd">cpm</span><span class="hl std">(dge_list_filter,</span><span class="hl kwc">log</span><span class="hl std">=T)</span> <span class="hl com">#log values of count per milllion of filtered genes</span>
<span class="hl std">dge_list_filter_log</span><span class="hl kwb">&lt;-</span> <span class="hl kwd">as_tibble</span><span class="hl std">(dge_list_filter_log,</span><span class="hl kwc">rownames</span><span class="hl std">=</span><span class="hl str">'gene_name'</span><span class="hl std">)</span> <span class="hl com">#re format as data frame </span>
<span class="hl kwd">colnames</span><span class="hl std">(dge_list_filter_log)</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">'gene_name'</span><span class="hl std">,experiment_data</span><span class="hl opt">$</span><span class="hl std">sample_name)</span>
<span class="hl std">dge_list_filter_log_pivot</span><span class="hl kwb">&lt;-</span> <span class="hl kwd">pivot_longer</span><span class="hl std">(</span><span class="hl kwc">data</span><span class="hl std">=dge_list_filter_log,</span>
                                         <span class="hl kwc">cols</span> <span class="hl std">= CL08</span><span class="hl opt">:</span><span class="hl std">HS05,</span>
                                         <span class="hl kwc">names_to</span> <span class="hl std">=</span> <span class="hl str">'condition'</span><span class="hl std">,</span>
                                         <span class="hl kwc">values_to</span> <span class="hl std">=</span> <span class="hl str">'expression_level'</span><span class="hl std">)</span>
<span class="hl std">pl2</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ggplot</span><span class="hl std">(dge_list_filter_log_pivot,</span><span class="hl kwd">aes</span><span class="hl std">(</span><span class="hl kwc">x</span><span class="hl std">=condition,</span><span class="hl kwc">y</span><span class="hl std">=expression_level,</span><span class="hl kwc">fill</span><span class="hl std">=condition))</span><span class="hl opt">+</span>
  <span class="hl kwd">geom_violin</span><span class="hl std">(</span><span class="hl kwc">trim</span> <span class="hl std">= F)</span><span class="hl opt">+</span>
  <span class="hl com">#geom_boxplot(width=0.1,outlier.size = 0.4,outlier.alpha = 0.8)+</span>
  <span class="hl kwd">stat_summary</span><span class="hl std">(</span><span class="hl kwc">fun</span><span class="hl std">=</span><span class="hl str">'median'</span><span class="hl std">,</span><span class="hl kwc">size</span><span class="hl std">=</span><span class="hl num">0.1</span><span class="hl std">)</span><span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl std">(</span><span class="hl kwc">x</span><span class="hl std">=</span><span class="hl str">'sample name'</span><span class="hl std">,</span><span class="hl kwc">y</span><span class="hl std">=</span><span class="hl str">'log 2 expression'</span><span class="hl std">,</span>
       <span class="hl kwc">title</span><span class="hl std">=</span><span class="hl str">'log2 count per million'</span><span class="hl std">,</span>
       <span class="hl kwc">subtitle</span> <span class="hl std">=</span> <span class="hl str">'trimmed - no normalization'</span><span class="hl std">,</span>
       <span class="hl kwc">caption</span> <span class="hl std">=</span> <span class="hl kwd">Sys.time</span><span class="hl std">())</span><span class="hl opt">+</span>
  <span class="hl kwd">coord_flip</span><span class="hl std">()</span>
<span class="hl kwd">print</span><span class="hl std">(pl2)</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 10 rows containing missing values (geom_segment).
</pre></div>
</div><div class="rimage center"><img src="figure/diy-transcriptomics-Rhtmlauto-report-2.png" title="plot of chunk auto-report" alt="plot of chunk auto-report" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl com">#Normalization by TMM of filtered genes </span>
<span class="hl std">dge_list_filter_norm</span><span class="hl kwb">&lt;-</span><span class="hl kwd">calcNormFactors</span><span class="hl std">(dge_list_filter,</span><span class="hl kwc">method</span><span class="hl std">=</span><span class="hl str">'TMM'</span><span class="hl std">)</span>
<span class="hl kwd">head</span><span class="hl std">(dge_list_filter_norm,</span><span class="hl num">4</span><span class="hl std">)</span> <span class="hl com">#in samples tab we see the norm.factors not 1 like previous</span>
</pre></div>
<div class="output"><pre class="knitr r">## An object of class &quot;DGEList&quot;
## $counts
##           Sample1    Sample2    Sample3    Sample4    Sample5     Sample6     Sample7
## A1BG    752.04846   431.9503   358.5730   437.9121   550.6723   203.78881   270.08552
## A2M   40961.84373 35187.6535 21933.4878 27175.8787 53666.0276 11922.23965 31527.10930
## A2ML1 11747.57461   455.6885   635.3131   284.0064  3670.6829 15156.34034 14018.86090
## A2MP1    46.34474   113.3492   132.4680   124.3212   206.7615    31.90768    70.59246
##           Sample8     Sample9    Sample10
## A1BG    139.45345   87.214893  15.0434991
## A2M   19868.05351 7832.214117 673.6880095
## A2ML1 11454.19260 3620.340812 507.9121971
## A2MP1    43.36109    9.065524   0.6237514
## 
## $samples
##          group lib.size norm.factors
## Sample1      1 59329489    0.8882270
## Sample2      1 31241771    1.0443316
## Sample3      1 34214384    1.0166616
## Sample4      1 42422092    0.9615193
## Sample5      1 73038349    1.0069792
## Sample6      1 44991051    0.9875171
## Sample7      1 54660379    1.0239917
## Sample8      1 43264951    1.0599990
## Sample9      1 17605351    1.0302456
## Sample10     1  2352675    0.9917344
</pre></div>
<div class="source"><pre class="knitr r"><span class="hl std">dge_list_filter_norm_log</span><span class="hl kwb">&lt;-</span><span class="hl kwd">cpm</span><span class="hl std">(dge_list_filter_norm,</span><span class="hl kwc">log</span><span class="hl std">=T)</span>
<span class="hl std">dge_list_filter_norm_log</span><span class="hl kwb">&lt;-</span> <span class="hl kwd">as_tibble</span><span class="hl std">(dge_list_filter_norm_log,</span><span class="hl kwc">rownames</span><span class="hl std">=</span><span class="hl str">'gene_name'</span><span class="hl std">)</span>
<span class="hl kwd">colnames</span><span class="hl std">(dge_list_filter_norm_log)</span><span class="hl kwb">&lt;-</span><span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">'gene_name'</span><span class="hl std">,experiment_data</span><span class="hl opt">$</span><span class="hl std">sample_name)</span>
<span class="hl std">dge_list_filter_norm_log_pivot</span><span class="hl kwb">&lt;-</span> <span class="hl kwd">pivot_longer</span><span class="hl std">(</span><span class="hl kwc">data</span><span class="hl std">=dge_list_filter_norm_log,</span>
                                              <span class="hl kwc">cols</span> <span class="hl std">= CL08</span><span class="hl opt">:</span><span class="hl std">HS05,</span>
                                              <span class="hl kwc">names_to</span> <span class="hl std">=</span> <span class="hl str">'samples'</span><span class="hl std">,</span>
                                              <span class="hl kwc">values_to</span> <span class="hl std">=</span> <span class="hl str">'expression'</span><span class="hl std">)</span>
<span class="hl std">pl3</span><span class="hl kwb">&lt;-</span><span class="hl kwd">ggplot</span><span class="hl std">(dge_list_filter_norm_log_pivot,</span><span class="hl kwd">aes</span><span class="hl std">(</span><span class="hl kwc">x</span><span class="hl std">=samples,</span><span class="hl kwc">y</span><span class="hl std">=expression))</span><span class="hl opt">+</span>
  <span class="hl kwd">geom_violin</span><span class="hl std">(</span><span class="hl kwd">aes</span><span class="hl std">(</span><span class="hl kwc">fill</span><span class="hl std">=samples),</span><span class="hl kwc">trim</span><span class="hl std">=F)</span><span class="hl opt">+</span>
  <span class="hl kwd">stat_summary</span><span class="hl std">(</span><span class="hl kwc">fun</span><span class="hl std">=</span><span class="hl str">'median'</span><span class="hl std">,</span><span class="hl kwc">size</span><span class="hl std">=</span><span class="hl num">0.1</span><span class="hl std">)</span><span class="hl opt">+</span>
  <span class="hl kwd">labs</span><span class="hl std">(</span><span class="hl kwc">x</span><span class="hl std">=</span><span class="hl str">'sample name'</span><span class="hl std">,</span><span class="hl kwc">y</span><span class="hl std">=</span><span class="hl str">'log 2 expression'</span><span class="hl std">,</span>
       <span class="hl kwc">title</span><span class="hl std">=</span><span class="hl str">'log2 count per million'</span><span class="hl std">,</span>
       <span class="hl kwc">subtitle</span> <span class="hl std">=</span> <span class="hl str">'trimmed - normalized'</span><span class="hl std">,</span>
       <span class="hl kwc">caption</span> <span class="hl std">=</span> <span class="hl kwd">Sys.time</span><span class="hl std">())</span><span class="hl opt">+</span>
  <span class="hl kwd">coord_flip</span><span class="hl std">()</span>
<span class="hl kwd">print</span><span class="hl std">(pl3)</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 10 rows containing missing values (geom_segment).
</pre></div>
</div><div class="rimage center"><img src="figure/diy-transcriptomics-Rhtmlauto-report-3.png" title="plot of chunk auto-report" alt="plot of chunk auto-report" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl kwd">plot_grid</span><span class="hl std">(pl1, pl2, pl3,</span> <span class="hl kwc">labels</span> <span class="hl std">=</span> <span class="hl kwd">c</span><span class="hl std">(</span><span class="hl str">'A'</span><span class="hl std">,</span> <span class="hl str">'B'</span><span class="hl std">,</span> <span class="hl str">'C'</span><span class="hl std">),</span> <span class="hl kwc">label_size</span> <span class="hl std">=</span> <span class="hl num">12</span><span class="hl std">)</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning: Removed 10 rows containing missing values (geom_segment).

## Warning: Removed 10 rows containing missing values (geom_segment).

## Warning: Removed 10 rows containing missing values (geom_segment).
</pre></div>
</div><div class="rimage center"><img src="figure/diy-transcriptomics-Rhtmlauto-report-4.png" title="plot of chunk auto-report" alt="plot of chunk auto-report" class="plot" /></div><div class="rcode">
<div class="source"><pre class="knitr r"><span class="hl std">knitr</span><span class="hl opt">::</span><span class="hl kwd">stitch_rhtml</span><span class="hl std">(</span><span class="hl str">'diy_transcriptomics.R'</span><span class="hl std">)</span>
</pre></div>
<div class="warning"><pre class="knitr r">## Warning in file(con, &quot;r&quot;): cannot open file 'diy_transcriptomics.R': No such file or
## directory
</pre></div>
<div class="error"><pre class="knitr r">## Error in file(con, &quot;r&quot;): cannot open the connection
</pre></div>
</div></div>

  <p>The R session information (including the OS info, R version and all
    packages used):</p>

<div class="chunk" id="session-info"><div class="rcode"><div class="source"><pre class="knitr r">    <span class="hl kwd">sessionInfo</span><span class="hl std">()</span>
</pre></div>
<div class="output"><pre class="knitr r">## R version 4.0.3 (2020-10-10)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Pop!_OS 20.04 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8       
##  [4] LC_COLLATE=en_US.UTF-8     LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                  LC_ADDRESS=C              
## [10] LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods  
## [9] base     
## 
## other attached packages:
##  [1] cowplot_1.1.0             forcats_0.5.0             stringr_1.4.0            
##  [4] dplyr_1.0.2               purrr_0.3.4               readr_1.4.0              
##  [7] tidyr_1.1.2               tibble_3.0.4              ggplot2_3.3.2            
## [10] tidyverse_1.3.0           tximport_1.16.1           EnsDb.Hsapiens.v86_2.99.0
## [13] ensembldb_2.14.0          AnnotationFilter_1.14.0   GenomicFeatures_1.42.1   
## [16] AnnotationDbi_1.52.0      Biobase_2.50.0            GenomicRanges_1.42.0     
## [19] GenomeInfoDb_1.26.2       IRanges_2.24.1            S4Vectors_0.28.1         
## [22] BiocGenerics_0.36.0       edgeR_3.30.3              limma_3.44.3             
## 
## loaded via a namespace (and not attached):
##  [1] fs_1.5.0                    ProtGenerics_1.22.0         bitops_1.0-6               
##  [4] matrixStats_0.57.0          lubridate_1.7.9.2           bit64_4.0.5                
##  [7] progress_1.2.2              httr_1.4.2                  backports_1.2.1            
## [10] tools_4.0.3                 R6_2.5.0                    DBI_1.1.0                  
## [13] lazyeval_0.2.2              colorspace_2.0-0            withr_2.3.0                
## [16] tidyselect_1.1.0            prettyunits_1.1.1           bit_4.0.4                  
## [19] curl_4.3                    compiler_4.0.3              cli_2.2.0                  
## [22] rvest_0.3.6                 xml2_1.3.2                  DelayedArray_0.16.0        
## [25] rtracklayer_1.50.0          labeling_0.4.2              scales_1.1.1               
## [28] askpass_1.1                 rappdirs_0.3.1              digest_0.6.27              
## [31] Rsamtools_2.6.0             XVector_0.30.0              pkgconfig_2.0.3            
## [34] MatrixGenerics_1.2.0        dbplyr_2.0.0                highr_0.8.1                
## [37] readxl_1.3.1                rlang_0.4.9                 rstudioapi_0.13            
## [40] RSQLite_2.2.1               generics_0.1.0              farver_2.0.3               
## [43] jsonlite_1.7.2              BiocParallel_1.24.1         RCurl_1.98-1.2             
## [46] magrittr_2.0.1              GenomeInfoDbData_1.2.4      Matrix_1.2-18              
## [49] fansi_0.4.1                 Rcpp_1.0.5                  munsell_0.5.0              
## [52] Rhdf5lib_1.10.1             lifecycle_0.2.0             stringi_1.5.3              
## [55] SummarizedExperiment_1.20.0 zlibbioc_1.36.0             rhdf5_2.32.4               
## [58] BiocFileCache_1.14.0        grid_4.0.3                  blob_1.2.1                 
## [61] crayon_1.3.4                lattice_0.20-41             haven_2.3.1                
## [64] Biostrings_2.58.0           hms_0.5.3                   locfit_1.5-9.4             
## [67] knitr_1.30                  pillar_1.4.7                biomaRt_2.46.0             
## [70] reprex_0.3.0                XML_3.99-0.5                glue_1.4.2                 
## [73] evaluate_0.14.1             modelr_0.1.8                vctrs_0.3.6                
## [76] cellranger_1.1.0            gtable_0.3.0                openssl_1.4.3              
## [79] assertthat_0.2.1            xfun_0.19                   broom_0.7.3                
## [82] GenomicAlignments_1.26.0    tinytex_0.28                memoise_1.1.0              
## [85] ellipsis_0.3.1
</pre></div>
<div class="source"><pre class="knitr r">    <span class="hl kwd">Sys.time</span><span class="hl std">()</span>
</pre></div>
<div class="output"><pre class="knitr r">## [1] &quot;2020-12-21 16:11:30 IST&quot;
</pre></div>
</div></div>


</body>
</html>
